gateway/src/shard/impl.rs

```diff
@@ -180,6 +182,10 @@ impl Error for SendError {
 #[derive(Debug)]
 #[non_exhaustive]
 pub enum SendErrorType {
+    #[deprecated(
+        since = "0.9.0",
+        note = "ratelimiter.acquire_one() does not return an error anymore"
+    )]
```

Comment:
Since `CommandErrorType` and `SendErrorType` share members, it needs to be deprecated there, too. The `since` field should be the next patch version of twilight-gateway (main), as it will be removed in 0.9.0 (next). The note should probably explain the change in behaviour that matters to the user, something like "Ratelimiting no longer uses an actor, therefore cannot fail".

Reply:
If deprecating the whole `SendErrorType` enum this will add errors/deprecate warnings on `HeartbeaterNotStarted` etc. which we don't want to do?

Reply:
Sorry for my wording, I was referring to deprecating `CommandErrorType::ExecutorShutDown` (just like `SendErrorType::ExecutorShutDown`), not the whole enum